components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java

```diff
@@ -79,260 +56,95 @@
-        when(mockedScopeMgtDAO.isScopeExists(anyString(), anyInt())).thenThrow(new IdentityOAuth2ScopeServerException
-                ("dummyIdentityOAuth2ScopeServerException"));
-
-        oAuth2ScopeService.isScopeExists(name);
-    }
-
-    @Test
-    public void testDeleteScope() throws Exception {
-        String name = "dummyScopeName";
-        mockStatic(OAuthScopeCache.class);
-        when(OAuthScopeCache.getInstance()).thenReturn(mockedOAuthScopeCache);
-        when(mockedOAuthScopeCache.getValueFromCache(any(OAuthScopeCacheKey.class))).thenReturn(mockedScope);
-        doNothing().when(mockedScopeMgtDAO).deleteScopeByName(anyString(), anyInt());
-
-        oAuth2ScopeService.deleteScope(name);
-    }
-
-    @Test(expectedExceptions = IdentityException.class)
-    public void testDeleteScopeWithNullName() throws Exception {
-        oAuth2ScopeService.deleteScope(null);
-    }
-
-    @Test(expectedExceptions = IdentityException.class)
-    public void testDeleteScopeWithWrongName() throws Exception {
-        String name = "dumName";
-        mockStatic(OAuthScopeCache.class);
-        when(OAuthScopeCache.getInstance()).thenReturn(mockedOAuthScopeCache);
-        when(mockedOAuthScopeCache.getValueFromCache(any(OAuthScopeCacheKey.class))).thenReturn(null);
-        when(mockedScopeMgtDAO.isScopeExists(anyString(), anyInt())).thenReturn(false);
-
-        oAuth2ScopeService.deleteScope(name);
+    @DataProvider(name = "ProvideCacheConfigurations")
+    public static Object[][] provideCacheConfigurations() {
+        return new Object[][]{
+                {false},
+                {true}
+        };
     }
 
-    @Test(expectedExceptions = IdentityOAuth2ScopeServerException.class)
-    public void testDeleteScopeWithIdentityOAuth2ScopeServerException() throws Exception {
-        String name = "dummyScopeName";
-        mockStatic(OAuthScopeCache.class);
-        when(OAuthScopeCache.getInstance()).thenReturn(mockedOAuthScopeCache);
-        when(mockedOAuthScopeCache.getValueFromCache(any(OAuthScopeCacheKey.class))).thenReturn(mockedScope);
-        doThrow(new IdentityOAuth2ScopeServerException("dummyException")).when(mockedScopeMgtDAO).deleteScopeByName
-                (anyString(), anyInt());
-
-        oAuth2ScopeService.deleteScope(name);
+    @Test(dataProvider = "ProvideCacheConfigurations")
+    public void testGetScope(boolean existWithinCache) throws Exception {
+        Scope dummyScope = new Scope(SCOPE_NAME, SCOPE_DESCRIPTION);
+        oAuth2ScopeService.registerScope(dummyScope);
+        if (!existWithinCache) {
+            OAuthScopeCache.getInstance().clearCacheEntry(new OAuthScopeCacheKey(SCOPE_NAME, Integer.toString(
+                    Oauth2ScopeUtils.getTenantID())));
+        }
+        assertEquals(oAuth2ScopeService.getScope(SCOPE_NAME).getName(), SCOPE_NAME, "Retrieving registered scope is " +
+                "failed");
+        oAuth2ScopeService.deleteScope(SCOPE_NAME);
```

Comment:
May be we can call delete in @AfterMethod

Reply:
Hi Danushka, we can't do this in tear down method since it is not common for all method. For example when it execute a test written for deleteScope method, and at the tearDown if it tries to delete again it throws a NPE.